runtime/gc_realtime/Scheduler.cpp

```diff
@@ -184,12 +184,56 @@ MM_Scheduler::initialize(MM_EnvironmentBase *env)
 		return false;
 	}
 
+	if (_extensions->gcTrigger == 0) {
+		_extensions->gcTrigger = (_extensions->memoryMax / 2);
+		_extensions->gcInitialTrigger = (_extensions->memoryMax / 2);
+	}
+
+	_extensions->distanceToYieldTimeCheck = 0;
+
+	/* Only SRT passes this check as the commandline option to specify beatMicro is only enabled on SRT */
+	if (METRONOME_DEFAULT_BEAT_MICRO != _extensions->beatMicro) {
+		/* User-specified quanta time, adjust related parameters, unless they are used specified, too. */
+		if (METRONOME_DEFAULT_TIME_WINDOW_MICRO == _extensions->timeWindowMicro) {
```

Comment:
@dmitripivkine I moved this big block of code from Scheduler here, to fix a problem of re-calculating timeWindowMicro when beatMicro is set. But I also added this condition to somewhat allow changing timeWindowMicro, if that is set too.

Even with this condition, setting  timeWindowMicro is not fully correct, since we can't recognize if METRONOME_DEFAULT_TIME_WINDOW_MICRO was set by command line option or it was just left default. For example. 

For example, setting -Xgc:targetPausetime=2 -XXgc:timeInterval=60000  would actually set values 2 and 40000, but for any other timeInterval  value (for example 60001) it would work.

Since timeInterval is an unsupported option, I'm thinking to go with this simplified fix rather than introducing extra code (for example MM_UserSpecifiedParameterUDATA). 

Let me know what you think or if you have a better suggestion.

Reply:
Will something be recalculated differently if we skip `(METRONOME_DEFAULT_BEAT_MICRO != _extensions->beatMicro)` check and do calculation for default value as well? Ideally calculation on defaults should have the same result as hardcoded default.
Alternatively we can move calculation of `timeWindowMicro` in default case up outside this `if`. The default value for `METRONOME_DEFAULT_BEAT_MICRO` is `3000`, so `20 * 3000 = 60000`, which is `METRONOME_DEFAULT_TIME_WINDOW_MICRO`